src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/DeleteExtendedQueryTagDataByWatermarkRange.sql

```diff
@@ -0,0 +1,55 @@
+/**************************************************************/
+--
+-- STORED PROCEDURE
+--     DeleteExtendedQueryTagDataByWatermarkRange
+--
+-- FIRST SCHEMA VERSION
+--     57
+--
+-- DESCRIPTION
+--     Deletes the extended query tag data in the provided watermark range
+--
+-- PARAMETERS
+--     @startWatermark
+--         * The inclusive start watermark.
+--     @endWatermark
+--         * The inclusive end watermark.
+--     @dataType
+--         * the data type of extended query tag. 0 -- String, 1 -- Long, 2 -- Double, 3 -- DateTime, 4 -- PersonName
+--     @tagKey
+--         * The extended query tag key
+------------------------------------------------------------------------
+CREATE OR ALTER PROCEDURE dbo.DeleteExtendedQueryTagDataByWatermarkRange
+    @startWatermark BIGINT,
+    @endWatermark BIGINT,
+    @dataType TINYINT,
+    @tagKey INT
+AS
+BEGIN
+    SET NOCOUNT ON
+    SET XACT_ABORT ON
+
+    BEGIN TRANSACTION
+
+        DECLARE @imageResourceType TINYINT = 0
+
+        IF @dataType = 0
+            DELETE FROM dbo.ExtendedQueryTagString WHERE TagKey = @tagKey AND ResourceType = @imageResourceType AND Watermark BETWEEN @startWatermark AND @endWatermark
```

Comment:
I dont think we need watermark here. Watermark is not even present in the index

Reply:
Also what about partitionKey?

Reply:
The watermark is required so we can execute the delete in batches. In `GetExtendedQueryTagBatches.sql` we get all of the tag data that needs to be deleted, and split it into "watermark ranges". Each range contains an equal amount of items to delete, so we can execute the work in equal batches.

Reply:
I'm a bit confused about partition key. I know it exists to partition the data so customers can query per partition. But can they also delete for a specific partition only? I didn't see any logic in the existing Delete XQT about deleting per partition key.
Or is it about speeding up the query by specifying the partition key so we can use one of the indexes existing in these tables? All of these contain more details like SOPInstanceIds - if we are trying to hit an index we would need to get all of this data, right?

Reply:
Karthik helped me understand some more about PartitionKey.

Since XQTs are not partitioned APIs, users create and delete query tags across partitions. When we go to delete a query tag, we are deleting all of the tag values for every instance in each partition, and that is by design. Similarly, when a new query tag is created, it will index all instances with that tag across partitions.
So it sounds like we do not need to add partition key to this query, which makes sense since the existing delete xqt sproc does not include it.

However, we talked about adding a new clustered index on watermark to improve the performance of this query. I will go ahead and add that